InventoryIndexer/Model/IsProductSalable.php

```diff
@@ -0,0 +1,48 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryIndexer\Model;
+
+use Magento\Framework\Exception\LocalizedException;
+use Magento\InventorySalesApi\Api\IsProductSalableInterface;
+use Magento\InventorySalesApi\Model\GetStockItemDataInterface;
+
+/**
+ * @inheritDoc
+ *
+ * Lightweight implementation for Storefront application.
+ */
+class IsProductSalable implements IsProductSalableInterface
+{
+    /**
+     * @var GetStockItemDataInterface
+     */
+    private $stockItemData;
+
+    /**
+     * @param GetStockItemDataInterface $stockItemData
+     */
+    public function __construct(GetStockItemDataInterface $stockItemData)
+    {
+        $this->stockItemData = $stockItemData;
```

Comment:
I would prefer naming it `getStockItemData` rather than `stockItemData`. In order to have `{verb}->execute()` structure.